### PR TITLE
[office-js] [office-js-preview] [office-runtime] Clarify SSO API support for event-based add-ins

### PR DESCRIPTION
**Pull request by samantharamon**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: N/A
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/office-js/index.d.ts [hunk 1]
```diff
@@ -1459,9 +1459,15 @@ declare namespace Office {
          *
          * **Hosts**: Excel, OneNote, Outlook, PowerPoint, Word
          * 
-         * **Important**: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox. 
-         *
          * **Requirement set**: {@link https://docs.microsoft.com/javascript/api/requirement-sets/common/identity-api-requirement-sets | IdentityAPI 1.3}
+         * 
+         * **Important**: In Outlook, this API isn't supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+         * 
+         * **Note**: In an Outlook event-based activation add-in, this API is supported starting from Office on Windows version 2111 (build 14701.20000).
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1466]:
```suggestion
         * **Note**: In an Outlook event-based activation add-in, this API is supported in Outlook on Windows starting from version 2111 (build 14701.20000). 
```

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -1536,9 +1536,15 @@ declare namespace Office {
          *
          * **Hosts**: Excel, OneNote, Outlook, PowerPoint, Word
          * 
-         * **Important**: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox. 
-         *
          * **Requirement set**: {@link https://docs.microsoft.com/javascript/api/requirement-sets/common/identity-api-requirement-sets | IdentityAPI 1.3}
+         * 
+         * **Important**: In Outlook, this API isn't supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+         * 
+         * **Note**: In an Outlook event-based activation add-in, this API is supported starting from Office on Windows version 2111 (build 14701.20000).
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:1543]:
```suggestion
         * **Note**: In an Outlook event-based activation add-in, this API is supported in Outlook on Windows starting from version 2111 (build 14701.20000).
```

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -1536,9 +1536,15 @@ declare namespace Office {
...
-         * **Important**: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox. 
-         *
          * **Requirement set**: {@link https://docs.microsoft.com/javascript/api/requirement-sets/common/identity-api-requirement-sets | IdentityAPI 1.3}
+         * 
+         * **Important**: In Outlook, this API isn't supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+         * 
+         * **Note**: In an Outlook event-based activation add-in, this API is supported starting from Office on Windows version 2111 (build 14701.20000). 
+         * To retrieve an access token in older builds, use 
+         * {@link https://docs.microsoft.com/javascript/api/office-runtime/officeruntime.auth?view=common-js-preview#office-runtime-officeruntime-auth-getaccesstoken-member(1) | 
+         * OfficeRuntime.auth.getAccessToken} instead. For more information, see
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:1546]: I'm not sure if you can split the link like this - how will the asterisk be rendered or parsed?
```suggestion
         * {@link https://docs.microsoft.com/javascript/api/office-runtime/officeruntime.auth?view=common-js-preview#office-runtime-officeruntime-auth-getaccesstoken-member(1) | OfficeRuntime.auth.getAccessToken}
         * instead. For more information, see 
```

File: types/office-js/index.d.ts [hunk 4]
```diff
@@ -1459,9 +1459,15 @@ declare namespace Office {
...
-         * **Important**: In Outlook, this API is not supported if the add-in is loaded in an Outlook.com or Gmail mailbox. 
-         *
          * **Requirement set**: {@link https://docs.microsoft.com/javascript/api/requirement-sets/common/identity-api-requirement-sets | IdentityAPI 1.3}
+         * 
+         * **Important**: In Outlook, this API isn't supported if the add-in is loaded in an Outlook.com or Gmail mailbox.
+         * 
+         * **Note**: In an Outlook event-based activation add-in, this API is supported starting from Office on Windows version 2111 (build 14701.20000). 
+         * To retrieve an access token in older builds, use 
+         * {@link https://docs.microsoft.com/javascript/api/office-runtime/officeruntime.auth?view=common-js#office-runtime-officeruntime-auth-getaccesstoken-member(1) | 
+         * OfficeRuntime.auth.getAccessToken} instead. For more information, see
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js/index.d.ts:1469]:
```suggestion
         * {@link https://docs.microsoft.com/javascript/api/office-runtime/officeruntime.auth?view=common-js#office-runtime-officeruntime-auth-getaccesstoken-member(1) | OfficeRuntime.auth.getAccessToken}
         * instead. For more information, see 
```

File: types/office-js-preview/index.d.ts (hunk 3 above)
Comment (samantharamon) [reply types/office-js-preview/index.d.ts:1546]: I tested rendering this in the reference repo and verified that the link works.

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Pull request merged**